### PR TITLE
Update the template with helpers -> lib.nixvim

### PR DESCRIPTION
**Pull request by refaelsh**
This PR is in cointiunuce of this comment: https://github.com/nix-community/nixvim/pull/2115#discussion_r1739915637.

Closes #1982

### CODE REVIEW

File: plugins/TEMPLATE.nix [hunk 1]
```diff
@@ -1,11 +1,10 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
```
Comment (MattSturgeon) [inline plugins/TEMPLATE.nix:6]:
```suggestion
}:
let
  inherit (lib.nixvim) defaultNullOpts;
in
```

File: plugins/TEMPLATE.nix [hunk 2]
```diff
@@ -14,11 +13,11 @@ helpers.neovim-plugin.mkNeovimPlugin config {
 
   # Optionally, explicitly declare some options. You don't have to.
   settingsOptions = {
-    foo = helpers.defaultNullOpts.mkUnsignedInt 97 ''
+    foo = lib.nixvim.defaultNullOpts.mkUnsignedInt 97 ''
```
Comment (MattSturgeon) [inline plugins/TEMPLATE.nix:16]:
```suggestion
    foo = defaultNullOpts.mkUnsignedInt 97 ''
```

File: plugins/TEMPLATE.nix [hunk 3]
```diff
@@ -14,11 +13,11 @@ helpers.neovim-plugin.mkNeovimPlugin config {
...
       The best birth year.
     '';
 
-    great_feature = helpers.defaultNullOpts.mkBool false ''
+    great_feature = lib.nixvim.defaultNullOpts.mkBool false ''
```
Comment (MattSturgeon) [inline plugins/TEMPLATE.nix:20]:
```suggestion
    great_feature = defaultNullOpts.mkBool false ''
```

**Review by MattSturgeon — CHANGES_REQUESTED** [review]
Awesome, we've been working on this in #1982, but as that's stalled we can merge here and make any further changes in the future.

If you could apply the suggestions below and then rebase/squash so that your commit is titled ``"plugins/template: `helpers` -> `lib.nixvim`"`` or similar, that'd be great! :grin:

**Comment by MattSturgeon** [comment]
Actually, looking at #1982 it already includes all these changes and is only stalled on further changes that'd be nice to have.

I'll go ahead and merge the original PR, but thanks for addressing this regardless!

**Comment by refaelsh** [comment]
> I'll go ahead and merge the original PR, but thanks for addressing this regardless!

Sure. No problem.

**Pull request closed without merging**